plugins/inputs/prometheus/kubernetes.go

```diff
@@ -188,10 +191,16 @@ func (p *Prometheus) watchPod(ctx context.Context, clientset *kubernetes.Clients
 		},
 	})
 
+	if err != nil {
+		<-ctx.Done()
+		return err
+	}
+
 	informerfactory.Start(ctx.Done())
 	informerfactory.WaitForCacheSync(wait.NeverStop)
 
 	<-ctx.Done()
+	return err
```

Comment:
```suggestion
	return nil
```

I get that it is nil if we get this far, but it is a little weird to see error here. Thoughts?